test/e2e/singlecluster/e2e_test.go

```diff
@@ -686,6 +687,69 @@ var _ = ginkgo.Describe("Kueue", func() {
 			}, util.LongTimeout, util.Interval).Should(gomega.BeTrue())
 		})
 
+		ginkgo.It("Should run with prebuilt workload", func() {
+			var wl *kueue.Workload
+			ginkgo.By("Create the pebuilt workload and the job adopting it", func() {
+				sampleJob = (&testingjob.JobWrapper{Job: *sampleJob}).
+					Label(constants.PrebuiltWorkloadLabel, "prebuilt-wl").
+					Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"5s", "-termination-grace-period", "0s"}).
+					BackoffLimit(0).
+					Obj()
+				testingjob.SetContainerDefaults(&sampleJob.Spec.Template.Spec.Containers[0])
+
+				wl = testing.MakeWorkload("prebuilt-wl", ns.Name).
+					Queue(localQueue.Name).
+					PodSets(
+						*testing.MakePodSet("main", 1).Containers(sampleJob.Spec.Template.Spec.Containers[0]).Obj(),
+					).
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, wl)).Should(gomega.Succeed())
+				gomega.Expect(k8sClient.Create(ctx, sampleJob)).Should(gomega.Succeed())
+			})
+
+			createdWorkload := &kueue.Workload{}
+			wlLookupKey := client.ObjectKeyFromObject(wl)
+			createdJob := &batchv1.Job{}
+			jobLookupKey := client.ObjectKeyFromObject(sampleJob)
+
+			ginkgo.By("Verify the prebuilt workload is adopted by the job", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, jobLookupKey, createdJob)).To(gomega.Succeed())
+					g.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
+					g.Expect(wl.Spec.PodSets[0].Template.Spec.Containers).To(gomega.BeComparableTo(createdJob.Spec.Template.Spec.Containers), "Check the way the job and workload is created")
+					g.Expect(createdWorkload.OwnerReferences).To(gomega.ContainElement(
+						gomega.BeComparableTo(metav1.OwnerReference{
+							Name: sampleJob.Name,
+							UID:  sampleJob.UID,
+						}, cmpopts.IgnoreFields(metav1.OwnerReference{}, "APIVersion", "Kind", "Controller", "BlockOwnerDeletion"))))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Verify the job is running", func() {
+				expectJobUnsuspendedWithNodeSelectors(jobKey, map[string]string{
+					"instance-type": "on-demand",
+				})
+			})
+
+			ginkgo.By("Delete all pods", func() {
+				gomega.Expect(util.DeleteAllPodsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Await for jobs completion", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), createdWorkload)).To(gomega.Succeed())
+
+					g.Expect(createdWorkload.Status.Conditions).To(gomega.ContainElement(
+						gomega.BeComparableTo(metav1.Condition{
+							Type:    kueue.WorkloadFinished,
+							Status:  metav1.ConditionTrue,
+							Reason:  "JobFinished",
+							Message: "Job failed",
```

Comment:
Actually, I think this job may ether fail or succeed, so I'm concern it may flake. I think not asserting on the message would be enough.

The reason is that when SIGTERM is sent to the `"gcr.io/k8s-staging-perf-tests/sleep:v0.0.3` container it may exit with 0 if the trap is already registered: https://github.com/kubernetes/perf-tests/blob/83cc34c18d8b0a5583d3eec257a0c1bcc3cbd200/util-images/sleep/main.go#L55.  I think if the trap is not registered it will exit with 137 after SIGKILL. To ensure this is fast we also set `TerimnationGracePeriod(1).` here: https://github.com/kubernetes-sigs/kueue/blob/a68c95860549a85061ae53db73d32de4df3c3446/test/e2e/singlecluster/e2e_test.go#L124C6-L124C32. 

I know this is hacky, we may revisit how to do it better.

Reply:
done